src/task-actions.ts

```diff
@@ -291,34 +297,100 @@ export async function starknetVoyagerAction(args: TaskArguments, hre: HardhatRun
         throw new HardhatPluginError(PLUGIN_NAME, msg);
     }
     //If contract hasn't been verified yet, do it
-    let contractPath = args.path;
+    if (args.path) {
+        await handleSingleContractVerification(args.path, args.address, voyagerUrl, hre);
+    } else {
+        const splitPaths = args.paths.split(" ");
+        await handleMultiPartContractVerification(splitPaths, args.address, voyagerUrl, hre);
+    }
+}
+
+async function handleMultiPartContractVerification(
```

Comment:
These two functions should be refactored. There is a lot of common functionality like path checking and response checking.